kivy/ext/mpl/backend_kivyagg.py

```diff
@@ -0,0 +1,216 @@
+    # examples. Not all GUIs require explicit instantiation of a
+    # main-level app (egg backend_gtk, backend_gtkagg) for pylab
+    FigureClass = kwargs.pop('FigureClass', Figure)
+    thisFig = FigureClass(*args, **kwargs)
+    return new_figure_manager_given_figure(num, thisFig)
+
+
+def new_figure_manager_given_figure(num, figure):
+    """
+    Create a new figure manager instance for the given figure.
+    """
+    canvas = FigureCanvasKivyAgg(figure)
+    manager = FigureManagerKivyAgg(canvas, num)
+    return manager
+
+
+class FigureCanvasKivyAgg(FigureCanvasAgg, Widget):
+    """
+    The canvas the figure renders into. Calls the draw and print fig
+    methods, creates the renderers, etc...
+
+    Public attribute
+
+      figure - A Figure instance
+
+    Note GUI templates will want to connect events for button presses,
+    mouse movements and key presses to functions that call the base
+    class methods button_press_event, button_release_event,
+    motion_notify_event, key_press_event, and key_release_event. See,
+    e.g., backend_gtk.py, backend_wx.py and backend_tkagg.py
+    """
+
+    def __init__(self, figure, **kwargs):
+        if _debug:
+            print('FigureCanvasKivyAgg: ', figure)
+        #super(FigureCanvasKivyAgg, self).__init__(figure, **kwargs)
+        FigureCanvasAgg.__init__(self, figure)
+        Widget.__init__(self, **kwargs)
+        self.figure = figure
+        self.blit()
+
+    def draw(self):
+        """
+        Draw the figure using the renderer
+        """
+        FigureCanvasAgg.draw(self)
+        if self.blitbox is None:
+            l, b, w, h = self.figure.bbox.bounds
+            w, h = int(w), int(h)
+            buf_rgba = self.get_renderer().buffer_rgba()
+        else:
+            bbox = self.blitbox
+            l, b, r, t = bbox.extents
+            w = int(r) - int(l)
+            h = int(t) - int(b)
+            t = int(b) + h
+            reg = self.copy_from_bbox(bbox)
+            buf_rgba = reg.to_string_argb()
+        texture = Texture.create(size=(w, h))
+        texture.blit_buffer(buf_rgba, colorfmt='rgba', bufferfmt='ubyte')
+        texture.flip_vertical()
+        with self.canvas:
+            Rectangle(texture=texture, pos=self.pos, size=(w, h))
+        #renderer = RendererKivyAgg(self.figure.dpi)
+        #self.figure.draw(renderer)
+
+    # You should provide a print_xxx function for every file format
+    # you can write.
+
+    # If the file type is not in the base set of filetypes,
+    # you should add it to the class-scope filetypes dictionary as follows:
+    filetypes = FigureCanvasBase.filetypes.copy()
+    filetypes['foo'] = 'My magic Foo format'
+
+    def blit(self, bbox=None):
+        # If bbox is None, blit the entire canvas to the widget. Otherwise
+        # blit only the area defined by the bbox.
+        self.blitbox = bbox
+
+    def print_foo(self, filename, *args, **kwargs):
```

Comment:
This needs to be changed to save png, and it should use http://kivy.org/docs/api-kivy.core.image.html#kivy.core.image.Image.save.